cratedb_toolkit/io/mongodb/cdc.py

```diff
@@ -0,0 +1,62 @@
+"""
+Basic relaying of a MongoDB Change Stream into CrateDB table.
+
+Documentation:
+- https://github.com/daq-tools/commons-codec/blob/main/doc/mongodb.md
+- https://www.mongodb.com/docs/manual/changeStreams/
+- https://www.mongodb.com/developer/languages/python/python-change-streams/
+"""
+
+import logging
+
+import pymongo
+import sqlalchemy as sa
+from commons_codec.transform.mongodb import MongoDBCDCTranslatorCrateDB
+
+from cratedb_toolkit.util import DatabaseAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class MongoDBCDCRelayCrateDB:
+    """
+    Relay MongoDB Change Stream into CrateDB table.
+    """
+
+    def __init__(
+        self,
+        mongodb_url: str,
+        mongodb_database: str,
+        mongodb_collection: str,
+        cratedb_sqlalchemy_url: str,
+        cratedb_table: str,
+    ):
+        self.cratedb_adapter = DatabaseAdapter(cratedb_sqlalchemy_url, echo=True)
+        self.mongodb_client: pymongo.MongoClient = pymongo.MongoClient(mongodb_url)
+        self.mongodb_collection = self.mongodb_client[mongodb_database][mongodb_collection]
+        self.table_name = self.cratedb_adapter.quote_relation_name(cratedb_table)
+        self.cdc = MongoDBCDCTranslatorCrateDB(table_name=self.table_name)
+
+    def start(self):
+        """
+        Subscribe to change stream events, convert to SQL, and submit to CrateDB.
+        """
+        with self.cratedb_adapter.engine.connect() as connection:
+            connection.execute(sa.text(self.cdc.sql_ddl))
+            for sql in self.cdc_to_sql():
+                if sql:
+                    connection.execute(sa.text(sql))
+                    connection.execute(sa.text(f"REFRESH TABLE {self.table_name};"))
```

Comment:
Do you think it is applicable to invoke a `REFRESH TABLE` command after applying each DML operation from the CDC stream? I am adding it here to be "better safe than sorry", but maybe it can be handled more relaxed? Please advise!

Reply:
@proddata responded to my question on another channel. Thanks!

> What if an INSERT statement and an UPDATE statement concerning the same record would be processed in a quick succession?

I assume this is using a primary key? Further updates will need to be applied sequentially anyway and probably have the form:
```sql
UPDATE my_mongo_table SET my_value = ? WHERE pk_key = 'xyz'
```
and therefore would lookup the latest version of the record in the translog.

Reply:
Excellent. So it will be safe to omit in this case. Thank you for clarifying.

Reply:
Removed `REFRESH TABLE` with the most recent adjustments. Thanks again.